docs/content/SRS/SRS.md

```diff
@@ -157,7 +183,54 @@ _The following Software Requirements Specification for `tinymorph` is using [Vol
 
 ### 9.1 Functional Requirements
 
-[Insert your content here.]
+FR-1. The system shall generate text suggestions based on user input in real-time, allowing users to continuously write while receiving suggestions for the next words, sentences, or paragraphs.
+
+FR-2. The system shall provide users with manual control over text generation by enabling them to select steering options like tone, style, or creativity level.
+
+FR-3. Users shall be able to interact with generated text through a left-to-right feedback panel that lets them modify model behavior in real-time by providing feedback on individual suggestions.
+
+FR-4. Users shall be able to set preferences for tone, style, voice, and formality, which the system will apply to all future generations of text.
+
+FR-5. Users shall be able to save their preferred configurations as profiles, enabling them to switch between different writing styles or goals (e.g., academic writing, casual blog posts, fiction writing).
+
+FR-6. The system shall allow users to upload custom system prompts (e.g., "Write like Hemingway" or "Adopt a motivational tone") to influence the overall behavior of the model.
+
+FR-7. The system shall allow users to navigate through their text non-linearly by providing a visual map or tree view that displays key points, topics, or sections.
+
+FR-8. The system shall provide explanations and recommendations for adjusting each hyperparameter, allowing users to better understand how changes will affect the model's output.
+
+FR-9. For each generated text suggestion, the system shall present multiple alternatives, which users can choose from, modify, or combine to form the final text.
+
+FR-10. Users shall be able to click on a specific part of the text and select from a dropdown of alternative phrasings or sentence structures generated by the model.
+
+FR-11. The system shall offer version control features, enabling users to revert to previous versions of the document or compare different drafts side-by-side.
+
+FR-12. The system shall support an offline mode where users can continue to write and interact with the editor without internet access, using pre-downloaded or locally hosted language models for text generation.
```

Comment:
offline mode requires on-device inference, which might be something that we won't focus for now.

but good to have in the future solution or future implementation or sth.